deep/serializers.py

```diff
@@ -137,22 +132,29 @@ def get_fields(self, *args, **kwargs):
         return fields
 
 
-class TempClientIdMixin():
+class TempClientIdMixin(serializers.ModelSerializer):
     """
     ClientId for serializer level only, not storing to the database.
     """
     client_id = serializers.CharField(required=False)
 
+    def _get_temp_client_id(self, validated_data):
+        try:
+            self.Meta.model._meta.get_field('client_id')
+            return None
```

Comment:
This always returns None please confirm .

Reply:
Yes, we don't want to remove **client_id** from validated_data if the model has a **client_id** field. so we just return None here. But if the model doesn't have the field then it will return (with pop -> remove from **valdiated_data** -> not passed to model.create()/model.update()).

I will make sure to add some comments here in the next PR.

Reply:
I will send this on the next PR.

![2021-08-05-142555](https://user-images.githubusercontent.com/7059255/128320103-926b443c-41ee-42c6-9fe3-a9d66a9f6409.png)